src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -290,6 +290,21 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
             Assert.AreEqual(lhs, rhs);
         }
 
+
+        private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
```

Comment:
@jnm2 Using C# 6 features means somebody can't build the framework using VS2012 (or 2013?) It's likely we will soon enough change our coding standard to allow require VS2015+ or equivalent, but we haven't done that yet. Rather than starting the discussion here, let's just go back to the use of constants, as we do everywhere.

[FYI @immeraufdemhund is referring to the fact that we did already make that change for the NUnit Gui, but not for any other software.]